crates/router/src/connector/iatapay.rs

```diff
@@ -271,7 +275,13 @@ impl ConnectorIntegration<api::Authorize, types::PaymentsAuthorizeData, types::P
         &self,
         req: &types::PaymentsAuthorizeRouterData,
     ) -> CustomResult<Option<types::RequestBody>, errors::ConnectorError> {
-        let req_obj = iatapay::IatapayPaymentsRequest::try_from(req)?;
+        let connector_router_data = iatapay::IatapayRouterData::try_from((
+            &self.get_currency_unit(),
+            req.request.currency,
+            req.request.amount,
+            req,
+        ))?;
+        let req_obj = iatapay::IatapayPaymentsRequest::try_from(&connector_router_data)?;
```

Comment:
Similarly you need to do for Refunds function, Search for `api::execute` in the file and refer to the PR linked to your tagged issue and you would have an idea on how to implement

Reply:
Thanks for pointing this out. I've implemented this in the latest commit